### DISCUSSION TITLE
Difference between max_detection_points and max_det_per_image

### DISCUSSION THREAD

**Discussion by arekmula** [rwightman/efficientdet-pytorch#266] [Q&A] [answered]
Hi,
I'm trying to optimize my model as much as I can, and I'm wondering what's the difference between `self.max_detection_points` and `self.max_det_per_image` in `DetBenchPredict` and `DetBenchTrain` classes. As far as I can see, they are both used to select TopK predictions, but `max_detection_points` is used before NMS and `max_det_per_image` is used after NMS. 

Can they be set to the same value?

**Comment by rwightman** [comment, marked as answer]
@arekmula Not sure it'd be a good idea to set them equal, especially if you lower detection points into the 100 range like max_det_per_image, for the topk, it's operating on class scores across feature maps for all levels and achors flattened, that's a lot of points with quite a bit of redundancy going into NMS. I feel you want to have  a decent number of candidates otherwise some modest scoring (but valid) scores are likely to be tossed due to higher scoring candidates across all the feature levels using up the slots. It really does depend on your use case, dataset, tradeoffs though so experiment, for lower res models you can likely lower it more safely.

max_det_per_image will have less impact on performance, but if you know that there is definitely going to be a certain numbers of detections per image you can lower or raise it.
